### PR TITLE
add ir for udp route

### PR DESCRIPTION
**Pull request by zhaohuabing**
Add ir for udp route #641 
Signed-off-by: zhaohuabing <zhaohuabing@gmail.com>

### CODE REVIEW

**Review by Xunzhuo — DISMISSED** [review]
LGTM, thanks!

File: internal/ir/xds.go [hunk 1]
```diff
@@ -471,3 +492,36 @@ func (t TLSInspectorConfig) Validate() error {
...
+// UDPListener holds the UDP listener configuration.
+// +k8s:deepcopy-gen=true
+type UDPListener struct {
+	// Name of the UDPListener
+	Name string
+	// Address that the listener should listen on.
+	Address string
+	// Port on which the service can be expected to be accessed by clients.
+	Port uint32
+	// Destinations associated with TCP traffic to the service.
```
Comment (alex-berger) [inline internal/ir/xds.go:505]: Did you mean `UDP` instead of `TCP` here?

**Comment by arkodg** [comment]
apart from the TCP typo, looks good @zhaohuabing !
thanks for also adding the TCP Validate calls which was missed earlier :)

**Comment by youngnick** [comment]
Yeah, I'm sorry to be a blocker, but I think we need to write down a design for what UDP options we'll make available, with some explanation of the tradeoffs, before we go too much further here.

**Comment by danehans** [comment]
>  I think we need to write down a design for what UDP options we'll make available, with some explanation of the tradeoffs..

@zhaohuabing do you mind creating a .md for this and submitting a separate PR?

**Comment by zhaohuabing** [comment]
Okay, I'll do that.

**Comment by zhaohuabing** [comment]
> Okay, I'll do that.

https://github.com/envoyproxy/gateway/pull/685

**Comment by arkodg** [comment]
thanks for adding the design doc with https://github.com/envoyproxy/gateway/pull/685, imho this PR should be good to go

**Review by arkodg — APPROVED** [review]

**Pull request merged**